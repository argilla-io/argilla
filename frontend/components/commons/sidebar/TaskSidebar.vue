<template>
  <div>
    <SidebarMenu
      :dataset="dataset"
      @refresh="onRefresh"
      @showMetric="onShowSidebarInfo"
      @changeViewMode="onChangeViewMode"
    />
    <!-- TODO: Use media queries -->
    <SidebarPanel
      v-if="sidebarVisible"
      :dataset="dataset"
      :class="dataset.task"
    >
      <div v-show="sidebarInfoType === 'progress'">
        <component :is="currentTaskProgress" :dataset="dataset" />
      </div>
      <div v-show="sidebarInfoType === 'stats'">
        <component :is="currentTaskStats" :dataset="dataset" />
      </div>
    </SidebarPanel>
  </div>
</template>

<script>
import { mapActions } from "vuex";
import { DatasetViewSettings } from "@/models/DatasetViewSettings";
import SidebarMenu from "./SidebarMenu";
import SidebarPanel from "./SidebarPanel";
export default {
  components: { SidebarMenu, SidebarPanel },
  props: {
    dataset: {
      type: Object,
      required: true,
    },
  },
  data: () => ({
    sidebarInfoType: "progress",
    sidebarVisible: false,
    width: window.innerWidth,
  }),
  computed: {
    currentTask() {
      return this.dataset.task;
    },
    currentTaskProgress() {
      return this.currentTask + "Progress";
    },
    currentTaskStats() {
      return this.currentTask + "Stats";
    },
  },
  updated() {
    window.onresize = () => {
      this.width = window.innerWidth;
    };
  },
  methods: {
    ...mapActions({
      changeViewMode: "entities/datasets/changeViewMode",
      refresh: "entities/datasets/refresh",
    }),
    onRefresh() {
      this.refresh({
        dataset: this.dataset,
      });
    },
    async onChangeViewMode(value) {
      await this.changeViewMode({
        dataset: this.dataset,
        value: value,
      });
    },
    onShowSidebarInfo(info) {
      if (this.sidebarInfoType !== info) {
        this.sidebarVisible = true;
      } else {
        this.sidebarVisible = !this.sidebarVisible;
      }
      this.sidebarInfoType = info;
      DatasetViewSettings.update({
        where: this.dataset.name,
        data: {
          visibleMetrics: this.sidebarVisible,
        },
      });
    },
  },
};
</script>
