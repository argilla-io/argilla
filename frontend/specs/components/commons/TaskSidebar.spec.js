import { mount } from "@vue/test-utils";
import TaskSidebar from "@/components/commons/sidebar/TaskSidebar";

function mountSidebar() {
  return mount(TaskSidebar, {
    propsData: {
      dataset: {
        task: "TextClassification",
        viewSettings: {
          annotationEnabled: false,
        },
      },
    },
  });
}

describe("TaskSidebar", () => {
  let spy = jest.spyOn(console, "error");
  afterEach(() => spy.mockReset());

  test("Required property", () => {
    mount(TaskSidebar);
    expect(spy).toBeCalledWith(
      expect.stringContaining('[Vue warn]: Missing required prop: "dataset"')
    );
  });

  test("renders properly", () => {
    const wrapper = mountSidebar();
    expect(wrapper).toMatchSnapshot();
  });

  test("Show sidebar panel", async () => {
    const wrapper = mountSidebar();
    await wrapper.setData({ sidebarVisible: true });
    expect(wrapper).toMatchSnapshot();
  });
});
