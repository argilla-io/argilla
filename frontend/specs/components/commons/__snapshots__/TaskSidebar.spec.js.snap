// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskSidebar Show sidebar panel 1`] = `
<div>
  <div class="sidebar"><span><div class="sidebar__info"><p>Mode</p> <sidebar-button id="explore" icon="explore-view" tooltip="Explore" type="Mode"></sidebar-button><sidebar-button id="annotate" icon="annotate-view" tooltip="Annotate" type="Mode"></sidebar-button><sidebar-button id="labelling-rules" icon="labelling-rules-view" tooltip="Define rules" type="Mode"></sidebar-button></div></span><span><div class="sidebar__info"><p>Metrics</p> <sidebar-button id="progress" icon="progress" tooltip="Progress" type="Metrics"></sidebar-button><sidebar-button id="stats" icon="metrics" tooltip="Stats" type="Metrics"></sidebar-button></div></span><span><div class="sidebar__info"><p>Refresh</p> <sidebar-button id="refresh" icon="refresh" tooltip="Refresh" type="Refresh"></sidebar-button></div></span> </div>
  <aside class="sidebar explore TextClassification">
    <div class="sidebar__wrapper">
      <div class="sidebar__content">
        <div>
          <textclassificationprogress dataset="[object Object]"></textclassificationprogress>
        </div>
        <div style="display: none;">
          <textclassificationstats dataset="[object Object]"></textclassificationstats>
        </div>
      </div>
    </div>
  </aside>
</div>
`;

exports[`TaskSidebar renders properly 1`] = `
<div>
  <div class="sidebar"><span><div class="sidebar__info"><p>Mode</p> <sidebar-button id="explore" active-view="explore" icon="explore-view" tooltip="Explore" type="Mode"></sidebar-button><sidebar-button id="annotate" active-view="explore" icon="annotate-view" tooltip="Annotate" type="Mode"></sidebar-button><sidebar-button id="labelling-rules" active-view="explore" icon="labelling-rules-view" tooltip="Define rules" type="Mode"></sidebar-button></div></span><span><div class="sidebar__info"><p>Metrics</p> <sidebar-button id="progress" icon="progress" tooltip="Progress" type="Metrics"></sidebar-button><sidebar-button id="stats" icon="metrics" tooltip="Stats" type="Metrics"></sidebar-button></div></span><span><div class="sidebar__info"><p>Refresh</p> <sidebar-button id="refresh" icon="refresh" tooltip="Refresh" type="Refresh"></sidebar-button></div></span> </div>
  <!---->
</div>
`;
